src/Resources/contao/classes/FrontendTemplate.php

```diff
@@ -389,6 +390,16 @@ private function setCacheHeaders(Response $response)
 			$response->setSharedMaxAge($objPage->cache);
 		}
 
+		// Tag response
+		if (System::getContainer()->has('fos_http_cache.http.symfony_response_tagger'))
```

Comment:
I think this code does not belong here. We cannot not assume that the return value of `FrontendTemplate::getResponse()` is always a page, can we? The code should go into the `PageRegular` class IMHO.

Reply:
I think it's correct here. It's the old `FrontenTemplate::output()` that was only called once and sent the response to the browser. It's the same thing for `getResponse()` now and we also set all the other caching headers regarding the page here (like disabling if FE preview is active and stuff).

Reply:
The old `output()` method still exists and the `getResponse()` method is not its replacement! You might as well use a front end template for an API response, therefore we cannot be sure it is always a page.

Reply:
So why are all the other caching headers set there?

Reply:
Is this a serious question? An API or ESI response can have cache headers, which does not make them a page.

Reply:
Of course it's a serious question, I don't have time for non-serious questions. ESI requests should certainly not use `FrontendTemplate::getResponse()`. It would set caching headers from **the page configuration** which makes no sense. An ESI fragment should do exactly the opposite.

Reply:
I'm using `FrontendTemplate::getResponse` a lot, so @leofeyer is right with his concern. Though it looks like the method is incorrect and should not add the page cache information…

Reply:
> It would set caching headers from the page configuration which makes no sense

That's true …

Reply:
So I'm moving the relevant ones to `PageRegular`? Nobody can really re-use that but this concept has to be reworked for CMF routing anyway.

Reply:
No, you don't have to move them. It is all based on `$objPage`, so either we change this or we might as well leave everything as it is.